app/models/gql/gql.rb

```diff
@@ -1,337 +1,350 @@
+      end
     end
-  end
 
-  def update_future
-    instrument('gql.performance.future.update_future') do
-      scenario.inputs_before.each { |input, value| update_graph(future, input, value) }
-      scenario.inputs_future.each { |input, value| update_graph(future, input, value) }
+    def update_future
+      instrument('gql.performance.future.update_future') do
+        scenario.inputs_before.each { |input, value| update_graph(future, input, value) }
+        scenario.inputs_future.each { |input, value| update_graph(future, input, value) }
+      end
     end
-  end
 
-  # @param graph a Qernel::Graph or :present, :future
-  #
-  def update_graph(graph, input, value)
-    case graph
-    when :present then graph = present
-    when :future  then graph = future
+    # @param graph a Qernel::Graph or :present, :future
+    #
+    def update_graph(graph, input, value)
+      case graph
+      when :present then graph = present
+      when :future  then graph = future
+      end
+      graph.query(input, value)
     end
-    graph.query(input, value)
-  end
 
-  # Standard query without modifiers. Queries present and future graph.
-  #
-  # @param query [String, Gquery] the single query.
-  # @return [ResultSet] Result query, depending on its gql_modifier
-  #
-  def query_standard(query)
-    ResultSet.create [
-      [scenario.start_year, query_present(query)],
-      [scenario.end_year, query_future(query)]
-    ]
-  end
+    # Standard query without modifiers. Queries present and future graph.
+    #
+    # @param query [String, Gquery] the single query.
+    # @return [ResultSet] Result query, depending on its gql_modifier
+    #
+    def query_standard(query)
+      ResultSet.create [
+        [scenario.start_year, query_present(query)],
+        [scenario.end_year, query_future(query)]
+      ]
+    end
 
-  # @param query [String] The query
-  # @return [Float] The result of the present graph
-  #
-  def query_present(query)
-    present.query(query)
-  end
+    # @param query [String] The query
+    # @return [Float] The result of the present graph
+    #
+    def query_present(query)
+      present.query(query)
+    end
 
-  # @param query [String] The query
-  # @return [Float] The result of the future graph
-  #
-  def query_future(query)
-    future.query(query)
-  end
+    # @param query [String] The query
+    # @return [Float] The result of the future graph
+    #
+    def query_future(query)
+      future.query(query)
+    end
 
-  def inspect
-    if @scenario
-      "#<#{ self.class.name } #{ @scenario.area_code }@#{ @scenario.end_year }>"
-    else
-      "#<#{ self.class.name } (unknown scenario)>"
+    def inspect
+      if @scenario
+        "#<#{ self.class.name } #{ @scenario.area_code }@#{ @scenario.end_year }>"
+      else
+        "#<#{ self.class.name } (unknown scenario)>"
+      end
     end
-  end
 
-end
+    private
 
+    def apply_initializer_inputs(graph = nil)
+      unless graph
```

Comment:
Yeah, it seemed like a cool idea earlier, but now at second glance it looks weird!